code/datums/jobs/job/xenomorph.dm

```diff
@@ -39,6 +39,17 @@
 	parent.show_character_previews(image('icons/Xeno/castes/larva.dmi', icon_state = "Bloody Larva", dir = SOUTH))
 	return TRUE
 
+/datum/job/xenomorph/add_job_points(amount)
+	if(iscrashgamemode(SSticker.mode))
+		var/datum/game_mode/infestation/crash/crash_gamemode = SSticker.mode
+		var/jobpoint_difference = crash_gamemode.get_jobpoint_difference() + amount
+		// Only add job points if there is not enough xenos.
+		if(jobpoint_difference <= 0)
+			return
+		amount = min(amount, jobpoint_difference)
+
+	return ..()
```

Comment:
The only other solution that I have is to make a proc in `_game_mode.dm`/`crash.dm` that is called by `/proc/add_job_points` that will adjust the job points. It feels needless since Crash is the only gamemode that will be using it, but that is the gamemode handling the points. That okay with you?